src/5.0_to_5.1/createFilterWithFirstMatchingCube.ts

```diff
@@ -0,0 +1,30 @@
+import { stringify } from "@activeviam/activeui-sdk-5.0";
+import {
+  createFilter,
+  MdxExpression,
+  Filter,
+  Cube,
+} from "@activeviam/activeui-sdk-5.1";
+
+/**
+ * Returns a {@link Filter} based on the input `mdx`, with the first matching cube.
+ */
+export function createFilterWithFirstMatchingCube(
+  mdx: MdxExpression,
+  cubes: Cube[],
+): Filter {
+  for (const cube of cubes) {
+    try {
+      return createFilter(mdx, cube);
+    } catch (e) {
+      // The filter creation might be successful with another cube.
+      // Continue.
+    }
+  }
+
+  throw new Error(
+    `The following MDX does not represent a filter: ${stringify(mdx, {
```

Comment:
Is that accurate?

If I understand correctly, getting here means that `createFilter` threw for every cube. And `createFilter` can indeed throw if `mdx` doesn't represent a filter, but also if the coordinates specified in `mdx` don't "match" what is specified `cube`. Right?

Reply:
During the migration the client is expected to pass all of the cubes that they're using, through `servers.json`.
If there is no cube for which this filter can be created, then from their perspective it doesn't represent a filter imo.

Would you phrase things differently ?

Also, an alternative would be to transform it as a custom filter instead. Potentially with a message like "error while migrating filter {id} with mdx {mdx}" in the report.

What do you think ?

Reply:
> Would you phrase things differently ?

I guess I'd rather have a message closer to what's in the description:
```suggestion
    `No cube contains the hierarchies expressed in the following filter MDX: ${stringify(mdx, {
```

> an alternative would be to transform it as a custom filter instead

What is a custom filter exactly?

> Potentially with a message like "error while migrating filter {id} with mdx {mdx}" in the report.

Actually I think we should do that either way. If we throw here the error will be caught higher up, and the error will be added to the error report. So we might as well as be specific as possible and specify both the filter id and the mdx, right?

Reply:
Actually this function is low level and is agnostic of the `id`, and there is not even necessarily an `id` if it's just a filter from a dashboard or widget state.
But if it's a saved filter being migrated then indeed the error will be caught higher up and reported with the file id.